File: docs/components_page/components/fade/fade.R
library(dashBootstrapComponents)

fade <- div(
  list(
    dbcButton("Toggle fade",
      id = "fade-button", n_clicks = 0,
      className = "mb-3"
    ),
    dbcFade(
      dbcCard(
        dbcCardBody(
          p(
            "This content fades in and out",
            className = "card-text"
          )
        )
      ),
      id = "fade",
      is_in = FALSE,
      appear = FALSE,
    )
  )
)


app$callback(
  output("fade", "is_in"),
  list(
    input("fade-button", "n_clicks"),
    state("fade", "is_in")
  ),
  function(n, is_in) {
    if (n > 0) {
      return(!is_in)
    }
    return(FALSE)
  }
)

File: docs/components_page/components/fade/fade.jl
using DashBootstrapComponents, DashHtmlComponents

fade = html_div([
    dbc_button("Toggle fade", id = "fade-button", className = "mb-3", n_clicks = 0),
    dbc_fade(
        dbc_card(
            dbc_cardbody(html_p("This content fades in and out", className = "card-text")),
        ),
        id = "fade",
        is_in = false,
        appear = false,
    ),
]);

callback!(
    app,
    Output("fade", "is_in"),
    Input("fade-button", "n_clicks"),
    State("fade", "is_in"),
) do n, is_in
    return n > 0 ? is_in == 0 : false
end;
